### ISSUE TITLE
Using bidirectional_dynamic_rnn with Grid3LSTMCell Gives Outputs with Different Shapes

### ISSUE THREAD

**Issue by selcouthlyBlue** [tensorflow/tensorflow#16925]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Microsoft Windows 8.1
- **TensorFlow installed from (source or binary)**:
Binary
- **TensorFlow version (use command below)**:
1.4 cpu
- **Python version**: 
3.6
- **Bazel version (if compiling from source)**:
N/A
- **GCC/Compiler version (if compiling from source)**:
N/A
- **CUDA/cuDNN version**:
N/A
- **GPU model and memory**:
N/A

### Describe the problem
I'm not sure if that's how using `bidirectional_dynamic_rnn` with Grid3LSTMCell works but it's giving outputs that are of different shapes: the first one has three dimensions and the second one has four dimensions.

### Source code / logs
Here's the code:

```
import tensorflow as tf
from tensorflow.contrib import grid_rnn

class BidirectionalGridRNNCellTest(tf.test.TestCase):
    def setUp(self):
        self.num_features = 1
        self.time_steps = 1
        self.batch_size = 1
        tf.reset_default_graph()
        self.input_layer = tf.placeholder(tf.float32, [self.batch_size, self.time_steps, self.num_features])
        self.cell_fw = grid_rnn.Grid3LSTMCell(num_units=8)
        self.cell_bw = grid_rnn.Grid3LSTMCell(num_units=8)

    def test_bidirectional_dynamic_grid_rnn(self):
        outputs, output_states = tf.nn.bidirectional_dynamic_rnn(self.cell_fw, self.cell_bw, self.input_layer, dtype=tf.float32)
        print(outputs)


if __name__ == '__main__':
    tf.test.main()
```

And here's the output of the print statement:

`((<tf.Tensor 'bidirectional_rnn/fw/fw/transpose:0' shape=(1, 1, 8) dtype=float32>,), <tf.Tensor 'ReverseV2:0' shape=(1, 1, 1, 8) dtype=float32>)`

**Comment by selcouthlyBlue** [comment]
Actually, this is also applicable to `Grid1LSTMCell`, and `Grid2LSTMCell`.

**Comment by bignamehyp** [comment]
Does it work without XLA?

**Comment by selcouthlyBlue** [comment]
What do you mean by XLA?

**Comment by bignamehyp** [comment]
I can reproduce this issue. @nealwu can you please take a look or redirect? Thanks.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 106 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 121 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 136 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 151 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 166 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 181 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 196 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 211 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
